source/uwp/Renderer/lib/XamlHelpers.h

```diff
@@ -19,6 +19,47 @@ namespace AdaptiveNamespace
             return result;
         }
 
+       template<typename T, typename TInterface, typename C>
+       static HRESULT IterateOverVector(_In_ ABI::Windows::Foundation::Collections::IVector<T*>* vector, const boolean stopOnFailure, C iterationCallback)
```

Comment:
>IterateOverVector [](start = 22, length = 17)

Hmm... having two copies of this seems fraught with peril, but it's fine for now as long as we have a plan to consolidate. Maybe make sure to log an issue to address the remainder of the exception vs hresult issues before you check in so we're sure to come back to this.

Reply:
Good point. I've filed #2533 to track this.

---
In reply to: [264376536](https://github.com/Microsoft/AdaptiveCards/pull/2504#discussion_r264376536) [](ancestors = 264376536)

Reply:
> I think YES, it should engage the fallback (to the OpenUrl in this case). The error (the unknown element) bubbles up.

I've filed #2535 to track implementing this.